nixos/modules/system/boot/systemd.nix

```diff
@@ -101,7 +101,7 @@ let
       "dev-hugepages.mount"
       "dev-mqueue.mount"
       "sys-fs-fuse-connections.mount"
-      "sys-kernel-config.mount"
+      ] ++ (optional (!config.boot.isContainer) "sys-kernel-config.mount") ++ [
```

Comment:
It might be /this/ gets pre-mounted in some lxc installations, but not all containers?

Reply:
How to fix it?
Add `config.boot.isLXCContainer`?
Or configure/patch `systemd` for not stopping on `sys-kernel-config.mount` failure?

Reply:
I think it'd be best to have systemd work in both cases. We don't always know the container environment that's used to spin the image up - having systemd work in both environments might be a better idea.

Can you check if this is a problem in other distros as well? In that case, I'd propose sending a PR to upstream systemd.